src/main/kotlin/org/jetbrains/exposed/dao/UUIDEntity.kt

```diff
@@ -0,0 +1,15 @@
+package org.jetbrains.exposed.dao
+
+import org.jetbrains.exposed.sql.Column
+import java.util.UUID
+
+open class UUIDTable(name: String = "", columName: String = "id") : IdTable<String>(name) {
+  override val id: Column<EntityID<String>> = varchar(columName, 36)
```

Comment:
Hmm, I think you meant that it becomes binary in some cases, like Oracle. If you use Postgres for example, there is a native `uuid` type (that Exposed uses) that you'll lose if you do this. Maybe what needs fixing is the default type used by Exposed for UUID instead?